File: CS/XPO/NET_Framework/RuntimeDbChooser.Module/BusinessObjects/CustomLogonParameters.cs
using DevExpress.ExpressApp.DC;
using DevExpress.ExpressApp.Model;
using DevExpress.ExpressApp.Security;

namespace RuntimeDbChooser.Module.BusinessObjects {
    public interface IDatabaseNameParameter {
        string DatabaseName { get; set; }
    }
    [DomainComponent]
    public class CustomLogonParametersForStandardAuthentication : AuthenticationStandardLogonParameters, IDatabaseNameParameter {
        private string databaseName = MSSqlServerChangeDatabaseHelper.DefaultDatabaseName;
        [ModelDefault("PredefinedValues", MSSqlServerChangeDatabaseHelper.Databases)]
        public string DatabaseName {
            get { return databaseName; }
            set { databaseName = value; }
        }
    }
    [DomainComponent]
    public class CustomLogonParametersForActiveDirectoryAuthentication : IDatabaseNameParameter {
        private string databaseName = MSSqlServerChangeDatabaseHelper.DefaultDatabaseName;

        [ModelDefault("PredefinedValues", MSSqlServerChangeDatabaseHelper.Databases)]
        public string DatabaseName {
            get { return databaseName; }
            set { databaseName = value; }
        }
    }
}

File: VB/XPO/NET_Framework/RuntimeDbChooser.Module/BusinessObjects/CustomLogonParameters.vb
Imports DevExpress.ExpressApp.DC
Imports DevExpress.ExpressApp.Model
Imports DevExpress.ExpressApp.Security

Namespace RuntimeDbChooser.Module.BusinessObjects

    Public Interface IDatabaseNameParameter

        Property DatabaseName As String

    End Interface

    <DomainComponent>
    Public Class CustomLogonParametersForStandardAuthentication
        Inherits AuthenticationStandardLogonParameters
        Implements IDatabaseNameParameter

        Private databaseNameField As String = MSSqlServerChangeDatabaseHelper.DefaultDatabaseName

        <ModelDefault("PredefinedValues", MSSqlServerChangeDatabaseHelper.Databases)>
        Public Property DatabaseName As String Implements IDatabaseNameParameter.DatabaseName
            Get
                Return databaseNameField
            End Get

            Set(ByVal value As String)
                databaseNameField = value
            End Set
        End Property
    End Class

    <DomainComponent>
    Public Class CustomLogonParametersForActiveDirectoryAuthentication
        Implements IDatabaseNameParameter

        Private databaseNameField As String = MSSqlServerChangeDatabaseHelper.DefaultDatabaseName

        <ModelDefault("PredefinedValues", MSSqlServerChangeDatabaseHelper.Databases)>
        Public Property DatabaseName As String Implements IDatabaseNameParameter.DatabaseName
            Get
                Return databaseNameField
            End Get

            Set(ByVal value As String)
                databaseNameField = value
            End Set
        End Property
    End Class
End Namespace
